node/acp.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package node
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/acp"
+)
+
+type ACPType uint8
+
+const (
+	NoACPType    ACPType = 0
+	LocalACPType ACPType = 1
+)
+
+// ACPOptions contains ACP configuration values.
+type ACPOptions struct {
+	acpType ACPType
+
+	// Note: An empty path will result in an in-memory ACP instance.
+	path string
```

Comment:
question: would `target` make more sense? as path is specific to local acp, I wonder if this will cause confusion for remote acp as it might maybe have a target/endpoint or something to sourcehub, not necessarily a path?

Reply:
`path` is consistent with the root store equivalent, I have a preference for us to use the same terminology

Reply:
Right I get that, I am just wondering if path would still make sense for remote ACP?

Reply:
No idea, if we want to rename it then we can do though